### PR TITLE
Add support for availability

### PR DESCRIPTION
**Pull request by jotak**
This PR is to bring basic support of availability.

There's some upcoming changes coming soon to bring much more stuff: multiple series aggregated availability, better variable/templating support, ...

### CODE REVIEW

**Comment by mtho11** [comment]
Should we really be versioning the .map files?

**Comment by jotak** [comment]
Actually, I have no idea. They were there, so I kept them :)

The whole "dist" directory could be unversionned actually, but that means we should automate builds & releases

**Comment by burmanm** [comment]
I think we should version the builds & releases (or sync them with Hawkular-Metrics releases?) @stefannegrea do you use some automated Jenkins build etc to create versions (or could we, if not) ?

**Comment by jotak** [comment]
cf PR #8

**Comment by tsegismont** [comment]
@mtho11 this is how other Grafana datasource plugins are managed, and I followed the example to get this project started. Before we decide to get rid of the dist files in VCS, what is the status of requesting Grafana to include the project in the plugin repository? Maybe their system relies on the sources being present in VCS.

**Comment by welshstew** [comment]
+1 would like to see availability in this :)

**Comment by jotak** [comment]
Finally got this PR back. I had to rewrite it completely due to other refactoring.

File: src/queryProcessor.js [hunk 1]
```diff
@@ -67,7 +70,7 @@ export class QueryProcessor {
 
   rawQuery(target, postData) {
     let uri = [
-      target.type + 's',            // gauges or counters
+      this.typeResources[target.type],            // gauges or counters
```
Comment (tsegismont) [inline src/queryProcessor.js]: Update (or simply remove) comment?

File: src/queryProcessor.js [hunk 2]
```diff
@@ -84,7 +87,7 @@ export class QueryProcessor {
   rawQueryLegacy(target, range, metricIds) {
     return this.q.all(metricIds.map(metric => {
       let uri = [
-        target.type + 's',            // gauges or counters
+        this.typeResources[target.type],            // gauges or counters
```
Comment (tsegismont) [inline src/queryProcessor.js]: Update (or simply remove) comment?

File: src/queryProcessor.js (hunk 1 above)
Comment (jotak) [inline src/queryProcessor.js]: done!

**Pull request merged**